pkg/controller/atlasproject/ipaccess_list.go

```diff
@@ -20,14 +22,20 @@ import (
 type atlasProjectIPAccessList mongodbatlas.ProjectIPAccessList
 
 func (i atlasProjectIPAccessList) Identifier() interface{} {
+	// hack: Atlas adds the CIDRBlock in case IPAddress is specified in the response.
+	// This breaks the contract (one field per List) and doesn't allow to "merge" lists.
```

Comment:
It doesn't actually break the contract, since it's not mentioned anywhere that the API **_response_** may only have one of the three 🙂

Reply:
yeah.. but this breaks our "joining" logic which looks ad "IDs" of joined entities. In this case we have to "denormalize" .
But I'll change the comment, you are right